### PR TITLE
docs: Clarify include_task v import_tasks with conditionals

### LINKED ISSUE #31596: Clarify difference between conditional include_task and import_task
##### ISSUE TYPE

 - Documentation Report

##### COMPONENT NAME

`docs/docsite/rst/playbooks_conditionals.rst`

##### ANSIBLE VERSION

n/a

##### CONFIGURATION

n/a

##### OS / ENVIRONMENT

n/a

##### SUMMARY

There is an unclear explanation [here](http://docs.ansible.com/ansible/latest/playbooks_conditionals.html#applying-when-to-roles-imports-and-includes):

> When used with include_* tasks instead of imports, the conditional is applied _only_ to the include task itself and not any other tasks within the included file(s). A common situation where this distinction is important is as follows:
> 
> ```
> # include a file to define a variable when it is not already defined
> 
> # main.yml
> - include_tasks: other_tasks.yml
>   when: x is not defined
> 
> # other_tasks.yml
> - set_fact:
>     x: foo
> - debug:
>     var: x
> ```
> 
> In the above example, if import_tasks had been used instead both included tasks would have also been skipped. With include_tasks instead, the tasks are executed as expected because the conditional is not applied to them.

When it says "both included tasks would have also been skipped", that's untrue.  If x is undefined, the first task is run and the second task is skipped.  This can be confirmed by experiment:

~~~
==> main1.yml <==
- hosts: localhost
  tasks:
    - include_tasks: other_tasks.yml
      when: x is not defined

==> main2.yml <==
- hosts: localhost
  tasks:
    - import_tasks: other_tasks.yml
      when: x is not defined

==> other_tasks.yml <==
# other_tasks.yml
- set_fact:
    x: foo
- debug:
    var: x 
~~~

I suggest the final paragraph could be reworded more clearly, e.g.

"In the above example, if import_tasks had been used instead then both imported tasks would have the condition `when: x is not defined` separately applied to them.  This means that if x is initially undefined, the first task is run, but the second task is skipped because the first task has defined a value for x."

### PR DESCRIPTION
**Pull request by ianw**
##### SUMMARY

The example given here is currently wrong, as the ``set_fact`` task will
be executed when using ``import_tasks``.

Correct and clarify this situation by expanding out what is happening
in the easy-to-get-wrong import mode.

Add some additional links to ``group_by`` documentation and the main
import/include discussion.

Fixes: #31596

##### ISSUE TYPE
 - Bugfix Pull Request
 - Docs Pull Request

### CODE REVIEW

**Review by bcoca — APPROVED** [review]

File: docs/docsite/rst/user_guide/playbooks_conditionals.rst [hunk 1]
```diff
@@ -173,15 +173,17 @@ Or with a role::
            when: ansible_os_family == 'Debian'
 
 You will note a lot of 'skipped' output by default in Ansible when using this approach on systems that don't match the criteria.
-Read up on the 'group_by' module in the :doc:`modules` docs for a more streamlined way to accomplish the same thing.
+In many cases the ``group_by`` module (see :doc:`modules`) can be a more streamlined way to accomplish the same thing; see
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_conditionals.rst:176]: We are moving away from `:doc:` links, please use a `:ref:` link here - I suggested `:ref:'group_by_module'` so the user can jump straight to the module documentation for that specific module.

File: docs/docsite/rst/user_guide/playbooks_conditionals.rst [hunk 2]
```diff
@@ -190,8 +192,19 @@ tasks within the included file(s). A common situation where this distinction is
     - debug:
         var: x
 
-In the above example, if ``import_tasks`` had been used instead both included tasks would have also been skipped. With ``include_tasks``
-instead, the tasks are executed as expected because the conditional is not applied to them.
+This expands at include time to the equivalent of::
+
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_conditionals.rst:196]: +1 for including the logical equivalent here

File: docs/docsite/rst/user_guide/playbooks_conditionals.rst [hunk 3]
```diff
@@ -190,8 +192,19 @@ tasks within the included file(s). A common situation where this distinction is
...
+        x: foo
+      when: x is not defined
+    - debug:
+        var: x
+      when: x is not defined
+
+Thus if ``x`` is initially undefined, the ``debug`` task will be skipped.  By using ``include_tasks`` instead of ``import_tasks``,
+both tasks from ``other_tasks.yml`` will be executed as expected.
+
+For more information on the differences between ``include`` v ``importing`` see :ref:`playbooks_reuse`.
```
Comment (acozine) [inline docs/docsite/rst/user_guide/playbooks_conditionals.rst]: typo? I'd make this "the differences between `include` and `import` . . . "

**Review by acozine — CHANGES_REQUESTED** [review]
Thanks for improving the docs. I've noted a few places where this change could be made even better.

File: docs/docsite/rst/user_guide/playbooks_conditionals.rst (hunk 1 above)
Comment (ianw) [reply docs/docsite/rst/user_guide/playbooks_conditionals.rst:176]: Done

File: docs/docsite/rst/user_guide/playbooks_conditionals.rst (hunk 3 above)
Comment (ianw) [reply docs/docsite/rst/user_guide/playbooks_conditionals.rst]: Done

**Review by acozine — APPROVED** [review]

**Comment by acozine** [comment]
Thanks @ianw!

**Pull request merged**